setup.py

```diff
@@ -1,75 +1,91 @@
+here = abspath(dirname(__file__))
 
-markupsafe_installs = (
-            sys.version_info >= (2, 6) and sys.version_info < (3, 0)
-        ) or sys.version_info >= (3, 3)
+with open(join(here, 'README.rst')) as r:
+    readme = r.read()
 
-install_requires = []
+tests_require = ['nose >= 0.11']
+if v < (3,):
+    tests_require.append('mock')
 
-if markupsafe_installs:
-    install_requires.append('MarkupSafe>=0.9.2')
+setup(
+    name='Mako',
+    use_scm_version={
+        'version_scheme': 'guess-next-dev',
+        'local_scheme': 'dirty-tag',
+        'write_to': 'mako/_version.py'
+    },
 
-try:
-    import argparse
-except ImportError:
-    install_requires.append('argparse')
+    setup_requires=['setuptools-scm>1.5.4'],
 
-setup(name='Mako',
-      version=VERSION,
-      description="A super-fast templating language that borrows the \
+    description="A super-fast templating language that borrows the \
  best ideas from the existing templating languages.",
-      long_description=readme,
-      classifiers=[
-          'Development Status :: 5 - Production/Stable',
-          'Environment :: Web Environment',
-          'Intended Audience :: Developers',
-          'Programming Language :: Python',
-          'Programming Language :: Python :: 3',
-          "Programming Language :: Python :: Implementation :: CPython",
-          "Programming Language :: Python :: Implementation :: PyPy",
-          'Topic :: Internet :: WWW/HTTP :: Dynamic Content',
-      ],
-      keywords='templates',
-      author='Mike Bayer',
-      author_email='mike@zzzcomputing.com',
-      url='http://www.makotemplates.org/',
-      license='MIT',
-      packages=find_packages('.', exclude=['examples*', 'test*']),
-      tests_require=['nose >= 0.11', 'mock'],
-      test_suite="nose.collector",
-      zip_safe=False,
-      install_requires=install_requires,
-      extras_require={},
-      entry_points="""
-      [python.templating.engines]
-      mako = mako.ext.turbogears:TGPlugin
+    long_description=readme,
+    classifiers=[
+        'Development Status :: 5 - Production/Stable',
+        'Environment :: Web Environment',
+        'Intended Audience :: Developers',
+        'Programming Language :: Python',
+        'Programming Language :: Python :: 3',
+        "Programming Language :: Python :: Implementation :: CPython",
+        "Programming Language :: Python :: Implementation :: PyPy",
+        'Topic :: Internet :: WWW/HTTP :: Dynamic Content',
+    ],
+    keywords='templates',
+    author='Mike Bayer',
+    author_email='mike@zzzcomputing.com',
+    url='http://www.makotemplates.org/',
+    license='MIT',
+    packages=find_packages('.', exclude=['examples*', 'test*']),
+    tests_require=tests_require,
+    test_suite="nose.collector",
+    
+    zip_safe=False,
+    install_requires=[
+        'MarkupSafe>=0.9.2'
+    ],
+    extras_require={
+        ':python_version == "2.6"': [
```

Comment:
I'm not associated in any way with pypa or any of its developers. I do maintain Python(x, y) which is a binary Python distribution for Windows. Mako is part of it (some projects require it) and since I take care to fulfill all of its dependencies and execute its tests I have a vested interest in making my life a bit easier.